### PR TITLE
Config overhaul

### PR DESCRIPTION
**Pull request by TarikViehmann**
Simplify the configuration of the CLIPS-Executive by:
1. remove rigid loading of cx_bringup package content across the project in favor of a configurable 'agent_dir' location which should point to the share directory of the package that implements the agent.
1. getting rid of specs and spec management.
   This is not needed as users can simply create multiple yaml files
1. make configuration files configurable.
   The CLIPS-Executive now accepts parameters
   'clips_executive_config' and 'clips_features_manager_config'.
   The default for the former is a file in
   agent_dir/params/clips_executive.yaml.
   The default for the latter is the whatever the value of the former
   is.
1. reworking the initialization stages of the CLIPS-Executive.
   There are no multi-stage initializations anymore.
   Instead, first the config for the features is fetched in order to
   request initialization of all mentioned features.
   Then, the CLIPS-Executive internal file loading is handled.
   Finally, the user files are loaded. There is no need to explicitly
   separate this in stages, as the loading is sequential in nature
   anyways.
1. remove clips-dirs variables. They are not needed as now the dirs are
   always both the cx_clips_executive dir as well as the agent_dir.
   If users want more dirs, they can configure them from within CLIPS.
1. rename all variables to kebab to camel case.
1. changed API to load configs from clips, now specify the file (full
   path). Path substitutions @CX_DIR@ and @AGENT_DIR@ are available for
   convenience.
1. simplify the debug config. The debug level is removed as it was unused and debug levels can be achieved by using the appropriate routers. debug/enable was also kinda pointless, hence it is gone now.
1. watch levels are now managed by the clips_manager directly
1. logging to file is now optional and the file name is changed to human readable time-date format.

At some point i got annoyed by bogus outputs so i cleaned up a few.

This PR is based on https://github.com/fawkesrobotics/ros2-clips-executive/pull/7, hence review that one first.

### CODE REVIEW

File: cx_clips_executive/src/cx_clips_executive/clips/skills.clp [hunk 1]
```diff
@@ -76,8 +54,8 @@
 		  (bind ?action_params (str-cat ?action_params " " ?param-value)))
 	)
 	(bind ?id (sym-cat ?action-name "_" (gensym*)))
-	(printout logwarn "Calling mapped skill '" ?sks "'" crlf)
-	(printout logwarn "Calling skill '" ?action-name " " ?action_params"'" crlf)
+	(printout t "Calling mapped skill '" ?sks "'" crlf)
+	(printout t "of action '" ?action-name " " ?action_params"'" crlf)
```
Comment (snoato) [inline cx_clips_executive/src/cx_clips_executive/clips/skills.clp:58]: this printout is a bit wonky, why two lines?

File: cx_features/src/cx_features/ClipsFeaturesManager.cpp [hunk 2]
```diff
@@ -88,60 +93,60 @@ ClipsFeaturesManager::on_configure(const rclcpp_lifecycle::State &state) {
...
-            declare_parameter(feat_name + "." + feat_param + ".value",
+            declare_parameter("clips_features." + feat_name + "." + feat_param + ".value",
                               rclcpp::PARAMETER_BOOL_ARRAY);
           }
           feature_param_map[feat_param] =
-              get_parameter(feat_name + "." + feat_param + ".value");
+              get_parameter("clips_features." + feat_name + "." + feat_param + ".value");
         }
 
         /////////////////////////////
```
Comment (snoato) [inline cx_features/src/cx_features/ClipsFeaturesManager.cpp:147]: guess we can get rid of this comment line while we're at it

File: cx_features/src/cx_features/ClipsFeaturesManager.cpp [hunk 3]
```diff
@@ -88,60 +93,60 @@ ClipsFeaturesManager::on_configure(const rclcpp_lifecycle::State &state) {
...
```
Comment (snoato) [inline cx_features/src/cx_features/ClipsFeaturesManager.cpp:93]: see comment below

**Review by snoato — CHANGES_REQUESTED** [review]
Some questions and comments, otherwise great improvement. Thanks!

File: cx_clips_executive/src/cx_clips_executive/ClipsExecutive.cpp [hunk 4]
```diff
@@ -69,16 +65,18 @@ ClipsExecutive::on_configure(const rclcpp_lifecycle::State &state) {
       std::make_shared<cx::CLIPSEnvManagerClient>("clips_manager_client_cx");
   clips_agenda_refresh_pub_ = create_publisher<std_msgs::msg::Empty>(
       "clips_executive/refresh_agenda", rclcpp::QoS(100).reliable());
-
-  std::string cx_bringup_dir;
-  std::string cx_features_dir;
+  declare_parameter<std::string>("agent_dir", ament_index_cpp::get_package_share_directory("cx_bringup"));
```
[the remark this answers is not in the document]
Comment (TarikViehmann) [reply cx_clips_executive/src/cx_clips_executive/ClipsExecutive.cpp:68]: thanks!

File: cx_clips_executive/src/cx_clips_executive/ClipsExecutive.cpp [hunk 5]
```diff
@@ -178,18 +160,20 @@ ClipsExecutive::on_activate(const rclcpp_lifecycle::State &state) {
   clips_ = clips_env_manager_node_->envs_["executive"].env;
 
   clips_->reset();
-  // TODO: REMOVE LATER!
-  clips_->watch("all");
+  clips_->watch("facts");
```
[the remark this answers is not in the document]
Comment (TarikViehmann) [reply cx_clips_executive/src/cx_clips_executive/ClipsExecutive.cpp:165]: nope, thanks

File: cx_clips_executive/src/cx_clips_executive/clips/skills.clp (hunk 1 above)
Comment (TarikViehmann) [reply cx_clips_executive/src/cx_clips_executive/clips/skills.clp:58]: because it can be quite long

**Review by snoato — APPROVED** [review]
Great improvement! Thanks!

**Pull request merged**